src/main/java/cz/cvut/kbss/termit/persistence/dao/changetracking/ChangeTrackingContextResolver.java

```diff
@@ -54,8 +65,19 @@ private URI resolveTermVocabulary(Term term) {
         } else {
             return em.createNativeQuery("SELECT DISTINCT ?v WHERE { ?t ?inVocabulary ?v . }", URI.class)
                      .setParameter("inVocabulary",
-                                   URI.create(cz.cvut.kbss.termit.util.Vocabulary.s_p_je_pojmem_ze_slovniku))
+                             URI.create(cz.cvut.kbss.termit.util.Vocabulary.s_p_je_pojmem_ze_slovniku))
                      .setParameter("t", term).getSingleResult();
         }
     }
+
+    private URI resolveExistingChangeTrackingContext(URI vocabularyURI) {
+        Optional<URI> vocabularyContext = vocabularies.getVocabularyContext(vocabularyURI);
+        return vocabularyContext.map(uri -> em.createNativeQuery(
+                                                      "SELECT DISTINCT ?ctc WHERE { GRAPH ?vc { ?vc ?hasChangeTrackingContext ?ctc } }", URI.class)
+                                              .setParameter("vc", uri)
+                                              .setParameter("hasChangeTrackingContext",
+                                                      URI.create(
+                                                              cz.cvut.kbss.termit.util.Vocabulary.s_i_ma_kontext_sledovani_zmen))
+                                              .getSingleResult()).orElse(null);
```

Comment:
This will throw `NoResultException` if no result is found for the query. I don't see it  being handled anywhere.

Reply:
Also, I do not understand why you return `null` from an `Optional` and then do a null-check in the calling method. I would rather return the `Optional` and do an `orElseGet` or something like that.

Reply:
Yes, this should throw a custom exception.

Reply:
That is my bad with the `null`. It was much more complicated and I did not notice that when I simplified it. Will change to `Optional`.